lint.go

```diff
@@ -1396,6 +1398,70 @@ func (f *file) lintTimeNames() {
 	})
 }
 
+func (f *file) lintDurationUnits() {
+	f.walk(func(node ast.Node) bool {
+		e, ok := node.(ast.Expr)
+		if !ok {
+			return true
+		}
+
+		tv := f.pkg.typesInfo.Types[e]
+		if !f.pkg.isNamedType(tv.Type, "time", "Duration") {
+			return true
+		}
+		if tv.Value == nil {
+			return true
+		}
+		// OK for 0 to be unitless; non-constants are OK.
+		if x, ok := exact.Int64Val(tv.Value); !ok || x == 0 {
+			return true
+		}
+
+		// e has a constant value. What matters now is how it was defined.
+		// It definition might be spread across many expressions:
+		//
+		// const (
+		//   a = iota
+		//   b
+		//   c = b * time.Second
+		//   e = c * 10
+		// )
+		//
+		// Rather than attempt to reconstruct e's origins,
+		// just handle the common cases.
+		switch e.(type) {
+		default:
+			// Assume that any non-literal, non-call expression is ok.
+			// In particular, allow 3 * time.Second, which is an *ast.BinExpr.
+			// This means that we fail to flag cases like
+			// time.Duration(15) + time.Duration(30) and
+			// time.Sleep(x) where 'const x = 10'.
+			return false
+		case *ast.CallExpr:
+			// Descend into call expressions in order to catch time.Duration(15).
+			return true
+		case *ast.BasicLit:
+			// A basic literal cannot possibly use time package constants.
+		}
+
+		f.errorf(e, 0.8, category("time"), "time.Duration expression should use time package constants: %v", f.render(e))
```

Comment:
`time.Duration expression "%v" should use time package constants`

it reads a bit more naturally, and will make the test a little clearer